src/UI/Components/Diagram/helpers.ts

```diff
@@ -602,3 +604,31 @@ export const moveCellFromColliding = (graph: dia.Graph, cell: dia.Cell) => {
     }
   } while (isColliding);
 };
+
+/**
+ * Finds the inter-service relations for a the given service model or embedded entity.
+ *
+ * @param {ServiceModel | EmbeddedEntity | undefined} serviceModel - The service model or embedded entity to find inter-service relations for.
+ * @returns {string[] | undefined} An array of entity types that have inter-service relations with the given service model or embedded entity, or undefined if the service model is undefined.
+ */
+export const findInterServiceRelations = (
+  serviceModel: ServiceModel | EmbeddedEntity | undefined,
```

Comment:
why do we need to support undefined here? Can the serviceModel be undefined, if yes, why?